esx_service/cli/vmdkops_admin_test.py

```diff
@@ -135,16 +135,27 @@ def test_set_invalid_options(self):
 
     def test_set_invalid_options(self):
         self.assert_parse_error('set --options="acces=read-write"')
+        self.assert_parse_error('set --options="attach-as=persisten"')
 
     def test_set_no_options(self):
         self.assert_parse_error('set --volume=volume_name')
 
     def test_set(self):
-        args = self.parser.parse_args('set --volume=vol_name --options="access=read-only"'.split())
+        args = self.parser.parse_args('set --volume=vol_name@datastore --options="access=read-only"'.split())
         self.assertEqual(args.func, vmdkops_admin.set_vol_opts)
-        self.assertEqual(args.volume, 'vol_name')
+        self.assertEqual(args.volume, 'vol_name@datastore')
         self.assertEqual(args.options, '"access=read-only"')
 
+        args = self.parser.parse_args('set --volume=vol_name@datastore --options="attach-as=persistent"'.split())
+        self.assertEqual(args.func, vmdkops_admin.set_vol_opts)
+        self.assertEqual(args.volume, 'vol_name@datastore')
+        self.assertEqual(args.options, '"attach-as=persistent"')
+
+        args = self.parser.parse_args('set --volume=vol_name@datastore --options="attach-as=independent_persistent"'.split())
+        self.assertEqual(args.func, vmdkops_admin.set_vol_opts)
+        self.assertEqual(args.volume, 'vol_name@datastore')
+        self.assertEqual(args.options, '"attach-as=independent_persistent"')
+
```

Comment:
good stuff ! 
Please do add a few negative tests (wrong attach mode string) to make sure it is handled properly.
Also, a test for unicode or non-English char set passed to attach-as needs to be added.